pandas-stubs/core/frame.pyi

```diff
@@ -732,11 +732,7 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
         axis: Axis | None = ...,
         level: Level | None = ...,
         copy: _bool = ...,
-        fill_value=...,
-        method: FillnaOptions | None = ...,
-        limit: int | None = ...,
-        fill_axis: Axis = ...,
-        broadcast_axis: Axis | None = ...,
+        fill_value: Hashable | None = ...,
```

Comment:
This should be `Scalar | NAType | None` .  In the stubs, we want to promote "good" behavior, and `Hashable` is too wide.  In addition, the docs say "fill value: scalar, default np.nan" .  Maybe the pandas source should change??